swarm/bmt/bmt.go

```diff
@@ -81,18 +82,17 @@ type Hasher struct {
 	bmt  *tree     // prebuilt BMT resource for flowcontrol and proofs
 }
 
-// New creates a reusable Hasher
-// implements the hash.Hash interface
+// New creates a reusable BMT Hasher that
 // pulls a new tree from a resource pool for hashing each chunk
 func New(p *TreePool) *Hasher {
 	return &Hasher{
 		pool: p,
 	}
 }
 
-// TreePool provides a pool of trees used as resources by Hasher
-// a tree popped from the pool is guaranteed to have clean state
-// for hashing a new chunk
+// TreePool provides a pool of trees used as resources by the BMT Hasher.
+// A tree popped from the pool is guaranteed to have a clean state ready
+// for hashing a new chunk.
```

Comment:
Do I understand correctly:
- BMT is the hashing algorithm used to hash Swarm resources
- every Swarm resource is represented by one `tree`
- a `TreePool` is used to speed up BMT hashing by providing "precompiled" structures (like a ThreadPool)
- this means there is a maximum (`TreePool.Capacity`) of concurrent resources that a Swarm node can hash concurrently

Reply:
almost:
BMT is the hashing algorithm used to hash CHUNKS, .... maximum of concurrent CHUNK hasing operations...

the rest is correct